src/pkg/selectors/scopes.go

```diff
@@ -317,6 +317,27 @@ func reduce[T scopeT, C categoryT](
 			continue
 		}
 
+		// if the details entry has a locationRef specified, use those folders in place
+		// of the repoRef folders, so that scopes can match against the display names
+		// instead of container IDs.
+		if len(ent.LocationRef) > 0 {
+			pb, err := path.Builder{}.
```

Comment:
we may need to add a helper function to the path package for this. I think you want `path.UnescapeAndAppend(path.Split(path.TrimTrailingSlash(ent.LocationRef))...)` and we need to make sure that the folders were output with escaping, which I [don't think is happening right now](https://github.com/alcionai/corso/blob/27c1e5c51140b238c6c3cbc3df79b290e37704fe/src/pkg/path/resource_path.go#L210) (it should be making a Builder and calling `String()` for escaping) :/

May be best to temporarily make a new function for `EscapedFolder()` until we can reconcile the current calls to `Folder()` that assumes no escaping

Reply:
Is the necessity of escaping just for ensuring parity between builder.String() and path.EscapedFolder()?

Reply:
we need escaping so `Split()` works properly. Otherwise we can't differentiate between the joined versions of `["this", "is/", "a", "path"]` and `["this", "is", "", "a", "path"]`

I realize there's a debate we could have about the handling of empty elements, but right now the path package is setup such that `Split` [takes an escaped path](https://github.com/alcionai/corso/blob/27c1e5c51140b238c6c3cbc3df79b290e37704fe/src/pkg/path/path.go#L582) and constructors for `Builder` [drop empty elements](https://github.com/alcionai/corso/blob/27c1e5c51140b238c6c3cbc3df79b290e37704fe/src/pkg/path/path.go#L158)